sdk/storage/AzureStorageBlob/Source/DataStructures/StorageBlobClientOptions.swift

```diff
@@ -261,7 +261,7 @@ public struct DownloadBlobOptions: AzureOptions {
 }
 
 /// User-configurable options for the blob upload operations.
-public struct UploadBlobOptions: AzureOptions {
+public class UploadBlobOptions: AzureOptions, Codable {
```

Comment:
Again, simplest fix. Would also be appropriate to remain a struct but change `BlobTransfer.downloadOptions/.uploadOptions` to not be optional.

Reply:
Would it make sense to do the same for BlobProperties? I'm not clear why you resolved the weird issue above by changing the *BlobOptions structs into classes, but for BlobProperties you instead made the property non-optional and return an empty value.